src/Drupal/Commands/pm/PmCommands.php

```diff
@@ -279,6 +293,39 @@ public function addInstallDependencies($modules)
         return $todo;
     }
 
+    /**
+     * Run requirements checks on the modules.
+     *
+     * @param array $modules
+     *   List of modules to validate.
+     *
+     * @throws \Exception
+     *
+     * @see \drupal_check_module()
+     */
+    public function validateInstallModules(array $modules)
+    {
+        require_once DRUSH_DRUPAL_CORE . '/includes/install.inc';
+        foreach ($modules as $module) {
+            module_load_install($module);
+            $requirements = $this->getModuleHandler()->invoke($module, 'requirements', ['install']);
+            if (is_array($requirements) && drupal_requirements_severity($requirements) == REQUIREMENT_ERROR) {
+                $reasons = [];
+                foreach ($requirements as $id => $requirement) {
+                    if (isset($requirement['severity']) && $requirement['severity'] == REQUIREMENT_ERROR) {
+                        $message = $requirement['description'];
+                        if (isset($requirement['value']) && $requirement['value']) {
+                            $message = dt('@requirements_message (Currently using @item version @version)', ['@requirements_message' => $requirement['description'], '@item' => $requirement['title'], '@version' => $requirement['value']]);
+                        }
+                        $reasons[$id] = $message;
+                    }
+                }
+
+                throw new \Exception(sprintf("Unable to install module '%s' due to unmet requirement(s):%s", $module, "\n  - " . implode("\n  - ", $reasons)));
```

Comment:
Thanks. Given our long history of not validating this, lets give the user the option to proceed despite the validation error. See https://github.com/drush-ops/drush/blob/10.x/src/Commands/core/UpdateDBCommands.php#L54-L59 for an example from updatedb.

Reply:
All done, it'll default to `false` in order to fail by default.

Reply:
Lets default to continue like updatedb does (i.e. no exception). Otherwise we will have to wait for a major version for this.

Reply:
I've updated the logic and test to reflect that.

And yes, that makes sense since we want to maintain backwards compatibility and avoid breaking any existing integrations/tests which rely on it (but one could also argue that if the module being installed doesn't meet the platform requirements in the first place, there's something wrong with their integration, and the failure _should_ be the appropriate default behaviour, and should lead to them fixing it on their end, and opt out with `--no`).